packages/otp2-tile-overlay/src/index.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  ConfiguredCompany,
+  MapLocationActionArg,
+} from "@opentripplanner/types"
+import { Layer, Popup,  Source, useMap } from "react-map-gl"
+import { StationPopup } from "@opentripplanner/vehicle-rental-overlay"
+import { StopPopup } from "@opentripplanner/stops-overlay"
+import React, { useEffect, useState } from "react"
+// eslint-disable-next-line prettier/prettier
+import type { EventData } from "mapbox-gl"
+import { LAYER_PAINT } from "./util"
+
+const OTP2TileLayerWithPopup = ({
+  configCompanies,
+  id,
+  network,
+  setLocation,
+  setViewedStop,
+  type
+}: {
+  /**
+   * Optional configuration item which allows for customizing properties of scooter and
+   * bikeshare companies
+   */
+  configCompanies?: ConfiguredCompany[]
+  id: string
+  name?: string
+  /**
+   * If `network` is specified, the layer will be filtered to only show vehicles from
+   * that network
+   */
+  network?: string
+  /**
+   * A method fired when a stop is selected as from or to in the popup
+   */
+  setLocation: (location: MapLocationActionArg) => void
+  /**
+   * A method fired when the stop viewer is opened in the popup
+   */
+  setViewedStop: ({ stopId }: { stopId: string }) => void
+  type: string
+}): JSX.Element => {
+  const { current: map } = useMap()
+
+  // TODO: handle this complex type: it can be a stop, a station, and some extra fields too
+  const [clickedEntity, setClickedEntity] = useState<any>(null)
```

Comment:
With the recommendation of not displaying popups, this state should be removed and the setter becomes the callback function for the published entity click event.

Reply:
Have added a comment to refactor the popups into a single coherent popup with no prop-rewriting